pkg/version/version.go

```diff
@@ -57,7 +59,9 @@ func FromString(versionString string) CiliumVersion {
 
 // GetCiliumVersion returns a initialized CiliumVersion structure
 func GetCiliumVersion() CiliumVersion {
-	return FromString(Version)
+	cver := FromString(Version)
+	metrics.VersionMetric.WithLabelValues(cver.Version)
```

Comment:
As far as I can tell, most other usage of `WithLabelValues()` is followed by a verb function like `Observe()`, `Add()`, `Inc()` but this one doesn't. My reading of the `WithLabelValues()` function comment is that this will result in creating the label with the value 0:

> // It is possible to call this method without using the returned Gauge to only  
> // create the new Gauge but leave it at its starting value 0. See also the      
> // SummaryVec example.

Does this mean that the gauge shows up with a label named with the version, or with a label named `version` with empty contents, or does it actually show up with `version` = `vX.Y.Z`?

Also, I'm wondering if `GetCiliumVersion()` is the best place to put the metric increment or if it should be configured somewhere else (from the main daemon/ files somewhere?). It seems like this function is only called from the CLI, rather than from the main daemon binary:

```
$ git grep "GetCiliumVersion()"
cilium/cmd/version.go:  return version.GetCiliumVersion()
pkg/version/version.go:func GetCiliumVersion() CiliumVersion {
```

Reply:
Agreeing with joestringer, perhaps it would be better to set this in the metrics init functions. Either `init` itself (https://github.com/cilium/cilium/blob/master/pkg/metrics/metrics.go#L1054) or when we create the configuration (https://github.com/cilium/cilium/blob/master/pkg/metrics/metrics.go#L503).
More generally, we do have a `process_start_time_seconds` metric but it is a prometheus built-in (https://github.com/cilium/cilium/blob/master/pkg/metrics/metrics.go#L1055). If we could figure out how to label that it would be a natural spot for this info (on the assumption that the goal is to know when cilium-agents have upgraded etc. `process_start_time_seconds` would also give you when that instance started).

Reply:
Hi @joestringer! Apologies for the confusion, I misunderstood the docs for `WithLabelValues`. My intention was to submit a dummy metric that is labelled `cilium_version=x.y.z`.

If the metric should be placed elsewhere, I'd be more than happy to move it to somewhere more appropriate. My intention is to submit as a prometheus metric. I'm not sure how helpful it would be to display the metric in the CLI as `cilium version` is already an option.

Reply:
@raybejjani, thanks for the feedback! I'll update accordingly and move the metric.